enhancements/ci-enhancement.md

```diff
@@ -0,0 +1,73 @@
+# Summary
+
+This enhancement proposal is to improve the overall performance and functionality of the operator-sdk CI and testing. There are a number of issues currently plaguing our system, some of which can be handled by switching to a higher tier of compute service and some of which will require serious revisions to our testing process.
+
+# Motivations
+
+The primary motivation for these enhancements is to improve developer productivity. Much time is lost in slow CI runs, inconsistent flakes and failures in CI, and a general difficulty in debugging said CI runs. 
+
+The primary issues are the following:
+
+Extremely slow TravisCI runs
+
+Can take upwards of 30mins for a single set of e2e tests to execute
+
+Unit test coverage is subpar
+
+Increased unit testing would also allow us to alleviate some of the e2e testing load
+
+Testing is stitched together across a number of bash scripts
+
+Very difficult to debug specific points of failure
+
+No unified structure for e2e testing
+
+General flakes and failures
+
+Travis passes but GitHub does not allow merging
+
+Goveralls requires rebasing
+
+Random websites which are called during docs checks are down and fail entire runs
+
```

Comment:
Some ideas that might help you with:

````
**Performance Issue**
Note that can take upwards of 30mins for a single set of e2e tests to execute. 

**Unit test coverage of the project is low**

May would be possible to decrease the quantity of e2e tests which are the root cause of the performance issues and the reason for the tests take to long to be executed. 

**E2E test done with shell script**
Since the e2e test are done in the shell is hard to troubleshooting and because of this, we cannot use coveralls to check what is or not covered with them.   

**NOTE**> Open question here. by using env test and the modules  "github.com/onsi/ginkgo" and `	."github.com/onsi/gomega"` we will able to troobolshoting/debug the e2e tests? 

**Low Manutence Ability**
The tests do not follow a single standard which makes harder we kept then maintained.  

**General flakes and failures (intermittent and not frequent)**
- Travis passes but GitHub does not allow merging
- Timeout issues and 404 errors during docs checks
```